src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-login.zsh

```diff
@@ -3,6 +3,12 @@
 #   Licensed under the MIT License. See License.txt in the project root for license information.
 # ---------------------------------------------------------------------------------------------
 
+# Prevent recursive sourcing
+if [[ -n "$VSCODE_LOGIN_INITIALIZED" ]]; then
+	return
+fi
+export VSCODE_LOGIN_INITIALIZED=1
```

Comment:
That means 2 more environment variables in the env unfortunately. Would be nice to understand why it happens so we didn't need to do this.